src/pages/course/course.js

```diff
@@ -25,6 +26,19 @@ router.get(
     res.locals.courseNumber = courseRow.number
     res.locals.courseName = courseRow.course_name
     res.locals.course_instances = result.rows
+    // showcasing using the checks
+    if (await checks.isLoggedIn(req)) {
+      res.locals.test_perms = await (async () => {
+        return Promise.all(
+          result.rows.map(r =>
+            checks.staffHasPermissionsForCourseInstance(req, r.id)
+          )
+        )
+      })()
+    } else {
+      res.locals.test_perms = Array.from('false'.repeat(result.rows.length))
```

Comment:
I don't think this does exactly what you think it does:

```js
> Array.from('false'.repeat(5))
< ["f", "a", "l", "s", "e", "f", "a", "l", "s", "e", "f", "a", "l", "s", "e", "f", "a", "l", "s", "e", "f", "a", "l", "s", "e"]
```

Reply:
this is mostly so that it doesn't crash for no one logged in, I'll change it to `f` for now